src/main/java/com/odysseusinc/athena/service/impl/VocabularyServiceImpl.java

```diff
@@ -292,20 +293,36 @@ public void checkBundleAndSharedUser(AthenaUser user, DownloadBundle bundle){
     @Override
     public List<License> saveLicenses(AthenaUser user, List<Integer> vocabularyV4Ids, LicenseStatus status) {
 
-        Iterable<License> licenses = vocabularyV4Ids.stream()
-                .map(vocabularyV4Id -> new License(user, new VocabularyConversion(vocabularyV4Id), status))
-                .collect(toList());
-        List<License> result = new ArrayList<>();
-        licenseRepository.save(licenses).iterator().forEachRemaining(result::add);
+        final List<License> newLicenses = buildLicenses(user, vocabularyV4Ids, status);
+        final List<License> savedLicenses = licenseRepository.save(newLicenses);
 
         conceptService.invalidateGraphCache(user.getId());
-        return result;
+        return savedLicenses;
+    }
+
+    @Override
+    public List<License> saveLicenses(AthenaUser user, List<License> licenses) {
+
+        final List<License> savedLicenses = licenseRepository.save(licenses);
+
+        conceptService.invalidateGraphCache(user.getId());
+        return savedLicenses;
+    }
+
+    private List<License> buildLicenses(AthenaUser user, List<Integer> vocabularyV4Ids, LicenseStatus status) {
```

Comment:
`private` methods should be at the end of the class

Reply:
ok